test/decode-text.test.js

```diff
@@ -0,0 +1,32 @@
+// FILEPATH: /home/rommel/workspace/contributions/busboy/test/decodeText.test.js
+
+const { test } = require('tap')
+const decodeText = require('../lib/utils/decodeText')
+
+test('decodeText', t => {
+  const testCases = [
+    { description: 'UTF-8 encoding', text: Buffer.from('Hello, World!', 'utf8'), initialCharset: 'utf8', outputCharset: 'utf8', expected: 'Hello, World!' },
+    { description: 'UTF-8 encoding empty', text: Buffer.from('', 'utf8'), initialCharset: 'utf8', outputCharset: 'utf8', expected: '' },
+    { description: 'Latin1 encoding', text: Buffer.from('Hello, World!', 'latin1'), initialCharset: 'latin1', outputCharset: 'latin1', expected: 'Hello, World!' },
+    { description: 'Latin1 encoding empty', text: Buffer.from('', 'latin1'), initialCharset: 'latin1', outputCharset: 'latin1', expected: '' },
+    { description: 'UTF-16LE encoding', text: Buffer.from('Hello, World!', 'utf16le'), initialCharset: 'utf16le', outputCharset: 'utf16le', expected: 'Hello, World!' },
+    { description: 'UTF-8 to UTF-16LE encoding', text: 'Hello, World!', initialCharset: 'utf16le', outputCharset: 'utf16le', expected: 'Hello, World!' },
+    { description: 'UTF-16LE encoding empty', text: Buffer.from('', 'utf16le'), initialCharset: 'utf16le', outputCharset: 'utf16le', expected: '' },
```

Comment:
Maybe use some utf16le characters instead of ascii characters to really ensure that the decoding works. Internally all strings are UCS2 so the string based comparison is not proving that the encoded string was decodee properly.

Reply:
This is enought as test with special char?

Reply:
Perfect! Lets run the tests